src/annotation/ruler.js

```diff
@@ -180,4 +173,55 @@ export default function () {
         });
       }
     });
-}
+
+export const rulerLabelVerticalSeparate = (cAcc) => (g) => {
+  const THRESHOLD = 2;
+  let labelBounds = [];
+
+  // Reset vertical shift
+  g.selectAll("text").each(function () {
+    select(this).attr("y", "");
+  });
+
+  // Calculate bounds
+  g.selectAll(".sszvis-ruler__label").each(function (d) {
+    const bounds = this.getBoundingClientRect();
+    labelBounds.push({
+      category: cAcc(d),
+      top: bounds.top,
+      bottom: bounds.bottom,
+      dy: 0,
+    });
+  });
+
+  // Sort by vertical position (only supports labels of same height)
+  labelBounds = labelBounds.sort((a, b) => ascending(a.top, b.top));
```

Comment:
```suggestion
  labelBounds = labelBounds.sort((a, b) => ascending(a.top, b.top));
```

sort function mutates then array in-place, no need to assign.